user/tests/wiring/no_fixture_i2c/i2c.cpp

```diff
@@ -105,13 +124,11 @@ test(I2C_02_test_acquire_wire1_buffer) {
     freeWireBuffers(&wire1LargerBuffers);
 
     assertEqual(hal_i2c_init(HAL_I2C_INTERFACE2, nullptr), (int)SYSTEM_ERROR_INVALID_ARGUMENT);
-
-    hal_i2c_end(HAL_I2C_INTERFACE2, nullptr);
 }
 #endif
 
 #if HAL_PLATFORM_EXTERNAL_RTC
-test(I2C_04_long_read_rtc)
+test(I2C_03_long_read_rtc)
 {
     // Test > 32 byte reads by reading from AM18x5 RTC peripheral
```

Comment:
This needs to look at `HAL_PLATFORM_EXTERNAL_RTC_I2C`, right?

Reply:
It does. if `HAL_PLATFORM_EXTERNAL_RTC` is defined, then `HAL_PLATFORM_EXTERNAL_RTC_I2C` must also be defined by these checks in `hal_platform.h`

```
#if HAL_PLATFORM_EXTERNAL_RTC
# ifndef HAL_PLATFORM_EXTERNAL_RTC_I2C
#  error "HAL_PLATFORM_EXTERNAL_RTC_I2C is not defined"
# endif /* HAL_PLATFORM_EXTERNAL_RTC_I2C */
# ifndef HAL_PLATFORM_EXTERNAL_RTC_I2C_ADDR
#  error "HAL_PLATFORM_EXTERNAL_RTC_I2C_ADDR is not defined"
# endif /* HAL_PLATFORM_EXTERNAL_RTC_I2C_ADDR */
# ifndef HAL_PLATFORM_EXTERNAL_RTC_CAL_XT
#  define HAL_PLATFORM_EXTERNAL_RTC_CAL_XT (0)
# endif /* HAL_PLATFORM_EXTERNAL_RTC_CAL_XT */
#endif /* HAL_PLATFORM_EXTERNAL_RTC */
```
I use these defines in the test itself
```
    hal_i2c_interface_t wireInterface = HAL_PLATFORM_EXTERNAL_RTC_I2C;
    uint8_t rtcAddress = HAL_PLATFORM_EXTERNAL_RTC_I2C_ADDR;
```

Reply:
I guess the only thing then is removing this `Wire1.begin()` then: https://github.com/particle-iot/device-os/pull/2559/files/98a9bd5aee8f26bcea23c47ead5e206b8a3a1285#diff-bf15c43f7b50354a9dee0d64fbb7cd3bd2b5bfcfd706d009369069cce9f4cfddR134

Reply:
Ah good point, I will replace it with `hal_i2c_init()` / `hal_i2c_begin()` using the defined interface